CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/ui/screens/home/HomeViewModel.kt

```diff
@@ -1,5 +1,44 @@
 package co.nimblehq.coroutine.ui.screens.home
 
+import androidx.hilt.lifecycle.ViewModelInject
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import co.nimblehq.coroutine.domain.data.entity.UserEntity
+import co.nimblehq.coroutine.domain.usecase.ChatUseCaseResult
+import co.nimblehq.coroutine.domain.usecase.GetUserUseCase
+import co.nimblehq.coroutine.lib.Event
 import co.nimblehq.coroutine.ui.base.BaseViewModel
+import kotlinx.coroutines.launch
 
-class HomeViewModel : BaseViewModel()
+interface Input {
+
+    fun users(): LiveData<List<UserEntity>>
+
+    fun showError(): LiveData<Event<String>>
+}
```

Comment:
I think the interface should be named `Output` instead, because we do not send any value or fetch anything, we are getting the value to observe it, right?
One more thing, can we change function `user()` and `showError()` to variable? We do not perform anything inside 😅 
```suggestion
interface Output {

    val users: LiveData<List<UserEntity>>

    val error: LiveData<Event<String>>
}
```

Reply:
> I think the interface should be named `Output` instead, because we do not send any value or fetch anything, we are getting the value to observe it, right?
> One more thing, can we change function `user()` and `showError()` to variable? We do not perform anything inside 😅

Ops 🙈 You're right @manh-t but I'm not sure about `val users: LiveData<List<UserEntity>>` It should be inside Input more than output 🤔 like we need to fetch data from API 🙏 What do you think?

Reply:
@Wadeewee 🤔  I think @manh-t's idea is the Fragment will observe new coming data from `users` and we don't use it for passing any value to VM so it should be in `Output` 😄  Ex: If we want to pass user-entered username and password to VM then it should be in `Input`.  Does it make sense to you? 😄

Reply:
@Wadeewee didn't you perform `fetchUsers()` internally? 😄  we only get result to use so putting `users` in `Output` makes sense imo 🙋

Reply:
@manh-t @hoangnguyen92dn Actually, I got the idea from HomeViewModel(RxJavaTemplate) but yeah I missed something 🙇‍♀️  What about If I still use Input to fetchUsers() but I will call inside init like this:
``` 
init {
        fetchUsers
    }
```
so for this case is it make sense for use Input, Please correct me 🙏

Reply:
@Wadeewee Let me explain the idea here 😄 In case we want to pass values from Fragment to VM, so it should be `Input` for VM, right? Then `Output` means the data from VM for Fragment observed. Back to our case, 

```
interface Input {

    fun users(): LiveData<List<UserEntity>>

    fun showError(): LiveData<Event<String>>
}
```

I think we don't use `users()` or `showError()` as the inputs due to we are not putting any values by using these function, so I think we could get rid of the `Input`, it could make confusion here 🤓 Then I would suggest to update with 2 approaches:

1. Create `HomeViewModel` Interface

interface HomeViewModel {
  val users: LiveData<List<UserEntity>>
  val errors: LiveData<Event<String>>
}

```
class HomeViewModelImpl @ViewModelInject constructor(
    private val getUserUseCase: GetUserUseCase
) : BaseViewModel(), HomeViewModel {
}
```

2.  Create public `users` and `errors` from VM

```
class HomeViewModel @ViewModelInject constructor(
    private val getUserUseCase: GetUserUseCase
) : BaseViewModel() {

    private val _users = MutableLiveData<List<UserEntity>>()
    private val _showError = MutableLiveData<Event<String>>()

    val users: LiveData<List<UserEntity>> = _users

    val errors: LiveData<Event<String>> = _showError
```

We could use `val` instead of `function` for `users` and `errors` as they are the properties from VM.

Does it make sense to you? 🙏

Reply:
I see @hoangnguyen92dn @manh-t 😄 Thanks for your explanation 🙇‍♀️ 🙏